### DISCUSSION TITLE
What's the best approach for integrating with an existing component library?

### DISCUSSION THREAD

**Discussion by mryechkin** [facebook/docusaurus#5752] [Q&A] [answered]
Hello! My team is looking to start using Docusaurus for documenting our React component library, and I'm looking for guidance on the best approach for this.

Currently we are using Storybook for generating documentation, and it lives in the same repo as the library source. We package the library code using Rollup, and then use Storybook to generate the static documentation site (that gets deployed separately). We also use Storybook for testing changes during development. All of our documentation is currently in MDX, but with Storybook-specific things.

What I'm looking for is to have something similar, but with Docusaurus. I'd like to continue using MDX for documentation, and be able to showcase the components, along with a prop table of some sort.

Our current project structure looks like this (simplified example):
```
library-root/
├─ docs/
│  ├─ GETTING-STARTED.md
│  ├─ SUPPORT.md
├─ src/
│  ├─ components/
│  │  ├─ Foo/
│  │  │  ├─ stories/
│  │  │  │  ├─ Foo.stories.mdx
│  │  │  ├─ Foo.js
│  │  │  ├─ index.js
│  ├─ hooks/
│  ├─ theme/
│  │  ├─ defaultTheme.js
│  │  ├─ index.js
│  ├─ index.js
```

Now I've run into some issues trying to add Docusaurus directly into the project at the root level (conflicts with `theme` folder, for example) - and I'm wondering if this is even the right approach, or if our docs site should be its own separate repo? And if that's the case, what would be your suggestions for how to test development changes using Docusaurus?

With Storybook we simply import components directly from `src` and then Webpack takes care of bundling it all when we run Storybook locally, but with Docusaurus I'm not quite sure how to have a similar workflow? Any help is greatly appreciated!

**Comment by slorber** [comment, marked as answer]
Hey,

Docusaurus is its own separate framework with different dependencies than your project. For example, you might use Webpack X and we might use Webpack Y. And you might have a single babel config file at the root that can only work for one or the other (not both). 

It would be better to have a monorepo/workspace and a package for each separate part of your project (website, storybook, component lib...)

By convention, Docusaurus looks into `src/theme` (hardcoded) and you can't disable it.

---

I'm not sure exactly to understand your use case.

> With Storybook we simply import components directly from src and then Webpack takes care of bundling it all when we run Storybook locally, but with Docusaurus I'm not quite sure how to have a similar workflow? Any help is greatly appreciated!

Docusaurus is a general tool, it's not only created to document frontend React component libraries. The community has probably built things to help document React components, but it's not something that is part of Docusaurus core.

**Reply by mryechkin** [reply]
Hey @slorber - thanks for following up.

Our use case is to use Docusaurus for showcasing all the available components in our component library, and the related documentation. I understand that it's not part of the standard core functionality, was just wondering how to go about integrating Docusaurus into our existing project, in general. But yes, the more I'm getting to know it, the more I understand that it would need to be a monorepo / workspace setup. I guess I was just looking for confirmation of that. Thank you!
